esx_service/utils/auth_data.py

```diff
@@ -538,18 +538,80 @@ def handle_upgrade_db_from_ver_1_0_to_ver_1_1(self):
         logging.error(error_msg)
         raise DbAccessError(self.db_path, error_msg)
 
+    def handle_upgrade_db_from_ver_1_1_to_ver_1_2(self):
+        """
+        Upgrade the db from version 1.1 to 1.2
+        In 1.1 the vms table had two columns namely vm_id and tenant_id
+        In 1.2 the vms table has three columns namely vm_id, tenant_id and vm_name
+        In the process of upgrading existing db from 1.1 to 1.2, we need to populate
+        the vm names for existing records. We try to populate them and keep None for vms
+        for which the name couldn't be found. The vmdk_ops admin code which tries to use
```

Comment:
I don't think we have a use case where user is upgrading from version 1.1 in X-ESX env. With the multi esx feature, the shared db would be from 1.2 version, and later on versions above. 

This is hit when the releases henceforth are deployed in single node mode and upgrades are done.

Does it answer your question?